tasks/task04/collect_tree_leaves.py

```diff
@@ -0,0 +1,32 @@
+""""
+You are given a tree-like data structure represented as nested dictionaries.
+Implement a function collect_leaves that accepts a tree and returns a list of all its leaves. A leaf is a bottom-most node in a tree.
+
+Implement a kind of unit tests via assert operator.
+"""
+from typing import Union
+
+
+def collect_leaves(u: Union[dict, list]):

```

Comment:
You could add typing for the return value since you added for typing for arguments.

Reply:
added